### PR TITLE
Tide: Don't create retests for unfinished or successful batch jobs

### PR DESCRIPTION
**Pull request by alvaroaleman**
Fixes https://github.com/kubernetes/test-infra/issues/12216

Makes Tide not create a new Batch job if a batch job of the same name for the same(org, repo, baseBranch, baseSHA, pullNumbers+PullHeadSHAs) already exists that:
* Is not finished
* Finished successfully

Diff is a lot smaller when disabling whitespace changes.

/assign @cjwagner @stevekuznetsov

### CODE REVIEW

File: prow/tide/tide_test.go [hunk 1]
```diff
@@ -1421,153 +1541,182 @@ func testTakeAction(clients localgit.Clients, t *testing.T) {
 		},
 	}
 
+	//logrus.SetLevel(logrus.DebugLevel)
```
Comment (matthyx) [inline prow/tide/tide_test.go:1544]: nit here?

**Comment by alvaroaleman** [comment]
> Could get @cjwagner eyes on this. Only comment is -- if we care about DRY or potentially future-proofing this if we change Refs in the future -- could we use Refs.String() as the key?

Not sure. `Refs.String()` does not e.G. contain the org and repo. Ppl changing it might not be aware of transient requirements incurred on it by using it that way. We know very precisely what we want here (reponame, basehash, headhashes) and thats IMHO very unlikely to change.

**Comment by alvaroaleman** [comment]
@stevekuznetsov are you okay with merging this by the end of the week if @cjwagner  doesn't find time to look into it till then? The worst case scenario for this pull is that we can not merge batches anymore

**Comment by alvaroaleman** [comment]
I'll roll this out and watch batches.
/hold cancel

**Pull request merged**